### ISSUE TITLE
Have detached proofs cross-reference their theorems

### ISSUE THREAD

**Issue by rbeezer** [PreTeXtBook/pretext#1678]
Have a detached proof function like an `xref`.

`<proof ref="a-big-theorem" text="type-global">...`

Somewhere early in the presentation of the proof, have a knowl/link to the theorem itself.

We could have the theorem link to multiple proofs by "discovering" the `ref` that point to it.

**Comment by arechnitzer** [comment]
+1. In proof/logic text I frequently want to give multiple proofs (perhaps even a couple of incorrect ones too) of a given statement, so the ref is important.

**Comment by davidfarmer** [comment]
There could be an option, maybe the default, for a cross-ref's proof to appear as

Proof of Theorem 3.14

**Comment by StevenClontz** [comment]
Since such proofs are not children of their theorem-like, they should probably be numbered, e.g.

> Proof 3.17 (of [Theorem 3.14](#))

where [Theorem 3.14](#) is a knowl, appearing when `@ref` is defined to be its `@xml:id`.

**Comment by rbeezer** [comment]
Good progress today on this - could be pretty efficient.

**Comment by rbeezer** [comment]
Exactly, but then we need to internationalize "of".  And there are two knowls in close proximity?

Been leaning towards

Proof (Theorem 3.14)

And still debating with myself...

But yes, @text is not necessary and will be a reasonable default.

**Comment by davidfarmer** [comment]
It is common and reasonable (maybe universal and reasonable) to not
number proofs.  If you go to the effort to make proofs numberable,
I suggest there should be an option to not number them, and that
should be the default.

**Comment by jjrsylvestre** [comment]
Couple of things:

1. Need to make it explicit in schema the difference between `INNER-PROOF` and `not(INNER-PROOF)` so that the latter can have `@ref` but the former should not.
2. There is currently a bug in knowl generation where the knowl file for the theorem being referenced is sometimes not generated, so that opening the theorem knowl from within the proof knowl returns a 404. Below is a minimal non-working example.

````
<?xml version="1.0" encoding="UTF-8" ?>
<pretext xmlns:xi="http://www.w3.org/2001/XInclude">

    <docinfo />

    <article xml:id="minimal">
        <title>A Minimal Article</title>

        <frontmatter>
            <titlepage>
                <author>
                    <personname>Robert A. Beezer</personname>
                    <institution>University of Puget Sound</institution>
                </author>
                <date><today /></date>
            </titlepage>
            <abstract>
                <p>Knowls in <tag>proof</tag> with <attr>ref</attr>.</p>
            </abstract>
        </frontmatter>

        <introduction>
            <p>Sometimes knowl file for target of <attr>ref</attr> in a <tag>proof</tag> doesn't get buld.</p>
        </introduction>

        <section xml:id="section-textual">
            <title>Just Some Text</title>
            <p>First a theorem.</p>
            <theorem xml:id="thm">
                <statement><p> A statement not involving a list. </p></statement>
            </theorem>
            <p>Now a <tag>proof</tag> with <attr>ref</attr> to the theorem above.</p>
            <proof ref="thm"><title>A proof</title>
                <p>Here is the text of the proof.</p>
            </proof>
        </section>
        
    </article>

</pretext>
````

**Comment by rbeezer** [comment]
Closing as implemented. A detached `<proof ref="theorem-id">` now doubles as a cross-reference to the theorem it proves — it's gathered alongside `xref`s (`$all-proofref` in `pretext-html.xsl`) and produces a knowl/link back to the theorem, exactly as proposed. The same applies to `argument`/`justification`/`reasoning`/`explanation` carrying `@ref`.

*Claude Opus 4.8 (1M context), acting as a review assistant for Rob Beezer*
